### ISSUE TITLE
Mattermost Desktop (Windows, client 5.1.0) crashes when changing screen resolutions

### ISSUE THREAD

**Issue by MikeSchlag** [mattermost/desktop#2159]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: https://docs.mattermost.com/install/troubleshooting.html](https://docs.mattermost.com/install/troubleshooting.html).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
When changing Mattermost location between multiple monitors, if it crosses a monitor/desktop that is a different resolution, the application crashes silently. Also will crash silently if previous resolution is not the same as before (changing monitor setup).

**Environment**
- Operating System: Windows 10
- Mattermost Desktop App version: 5.1.0 
- Mattermost Server version: 6.6.1 (on-premises server)

**Steps to reproduce**
Have a Windows 10 desktop with multiple monitors, set to different resolutions (e.g. #1 set to 1920x1080, #2 set to 3840x2160). Drag desktop client window from one screen to the other; somewhere between, the application will pause, then silently crash (no unhandled exception or other messages). Happens whether application is maximized and moved with mouse; un-maximized and moved with mouse; maximized and moved with Windows+left/right arrow keys; un-maximized and moved with Windows+left/right arrow keys.

Verifying monitor resolution issue: If all monitors are set to the same resolution (e.g. 1920x1080) then it can be moved in between monitors with no issues, with mouse or keyboard movements.

**Expected behavior**
Able to move client between multiple monitors at varying resolutions with no crash. This has worked on previous versions, back to at least 5.0.4 (last version I had before updating to 5.1.0).

**Observed behavior**

No messages are given, and no relevant log entries look to be present in main.log. Note that the main.log attached has had server and user ID info redacted manually by me prior to upload.

**Possible fixes**
Unknown
[main.log.bugreport.txt](https://github.com/mattermost/desktop/files/8921703/main.log.bugreport.txt)

**Comment by amyblais** [comment]
@jgilliam17 @lindy65 Are you able to reproduce this?

**Comment by jgilliam17** [comment]
I tried connecting my macbook to external monitor and was able to move desktop app around between two screens of different resolutions without a crash. 
I'll need a little bit of time to get a second monitor and check on win10.

**Comment by jgilliam17** [comment]
Tested on Win10, desktop app v5.1 and v5.0.4 using 2 monitors; 1st monitor resolution set at 2460x1080 and second at 1024x768
Able to move maximized/unmaximized app window using a mouse and/or keyboard shortcuts without experiencing a crash. 
Noting here that I had to adjust the size of the app afterwards, when going from larger to smaller monitor and vice versa  - I believe this is expected. @devinbinnie can correct me here if I'm wrong.

**Comment by lindy65** [comment]
@amyblais - I don't have an external monitor to test this

**Comment by tjdownes** [comment]
I can confirm this same behavior, Windows 11, using 3 of the same monitors. Just switching monitors does this to me, I don't even need different resolutions.

**Comment by tjdownes** [comment]
> Is there an `uncaughtException` file in your Mattermost settings folder where you found the logs? (should be one level up)

Would that be in `AppData\Roaming\Mattermost\logs`? if so, there's no such file, and `main.log` does not contain any related exceptions.

If there's some sort of debugging you need me to do, this is very repeatable. Just let me know what you need from me to get you the debug info you need.

**Comment by tjdownes** [comment]
Actually, I have several in the folder above `logs`, but none are even remotely related to the time period in question
![image](https://user-images.githubusercontent.com/273720/174623579-5e282a66-9391-462e-87b1-e9cc78197621.png)

**Comment by devinbinnie** [comment]
@tjdownes It's likely an Electron issue then related to your specific setup. Could you try on a different Windows machine (or VM) and see if you can still reproduce?

**Comment by tjdownes** [comment]
>Could you try on a different Windows machine (or VM) and see if you can still reproduce?

I'm not aware of any way to virtualize a multi-display setup. Do you have any documentation or suggestions on how to do that?

FYI, this is a pretty standard configuration. Nothing special here. 2060 RTX with studio drivers, Windows 11.

**Comment by tjdownes** [comment]
I tested a couple of other Electron apps yesterday, using 18.0.1 and 18.0.3 and could not produce the same issues with those apps.

**Comment by devinbinnie** [comment]
> > Could you try on a different Windows machine (or VM) and see if you can still reproduce?
> 
> I'm not aware of any way to virtualize a multi-display setup. Do you have any documentation or suggestions on how to do that?
> 
> FYI, this is a pretty standard configuration. Nothing special here. 2060 RTX with studio drivers, Windows 11.

I think VirtualBox will allow it. It might be better to try another machine.
I'm also using an NVIDIA card but with the GeForce drivers. Shot in the dark, but perhaps updating yours could help?

**Comment by tjdownes** [comment]
> perhaps updating yours could help?

I updated them yesterday :) Realistically the studio drivers should be more stable than the game-ready drivers, since the game-ready drivers are built on top of the studio drivers.

I've got a couple of teammates using MM with multiple monitors on Windows, so I will have them test it today and report if they have the same issues. I'll see if I can get my laptop connected to my monitors for testing as well.

**Comment by MikeSchlag** [comment]
Got the notification today that 5.1.1 was available. I just downloaded and installed, and the crashing is no longer present between screens of varying resolutions. Looks like it could probably be closed at this point. Thanks so much for fixing that!

**Comment by devinbinnie** [comment]
@tjdownes Are you seeing the same thing?

**Comment by tjdownes** [comment]
> @tjdownes Are you seeing the same thing?

YES! Thanks for the fix. It also fixed a bug I had not yet reported, where typing `/` to do actions resulted in a similar crash.

**Comment by devinbinnie** [comment]
Great! I'll close this issue then.
